lwgrp.spec

```diff
@@ -0,0 +1,194 @@
+%package -n liblwgrp0-openmpi3
+Summary: Light-weight Group Library for MPI process groups -- Shared libraries
+
+%description -n liblwgrp0-openmpi3
+Shared libraries for %{name}-openmpi3.
+%endif
+
+%package openmpi3-devel
+Summary: Light-weight Group Library for MPI process groups -- development files
+%if (0%{?suse_version} >= 1500)
+Requires: liblwgrp0-openmpi3%{_isa} = %version-%release
+%else
+Requires: %{name}-openmpi3%{_isa} = %version-%release
+%endif
+
+%description openmpi3-devel
+Development files for %{name}-openmpi3.
+%endif
+
+%if %{with_mpich}
+%package mpich
+Summary: Light-weight Group Library for MPI process groups
+BuildRequires: mpich-devel
+
+%description mpich
+The light-weight group library defines data structures and collective
+operations to group MPI processes as an ordered set.
+
+%if (0%{?suse_version} >= 1500)
+%package -n liblwgrp0-mpich
+Summary: Light-weight Group Library for MPI process groups -- Shared libraries
+
+%description -n liblwgrp0-mpich
+Shared libraries for %{name}-mpich.
+%endif
+
+%package mpich-devel
+Summary: Light-weight Group Library for MPI process groups -- development files
+%if (0%{?suse_version} >= 1500)
+Requires: liblwgrp0-mpich%{_isa} = %version-%release
+%else
+Requires: %{name}-mpich%{_isa} = %version-%release
+%endif
+
+%description mpich-devel
+Development files for %{name}-mpich.
+%endif
+
+%prep
+%setup -q
+%patch1 -p1
+autoreconf
+
+%build
+%global _configure ../configure
+export CC=mpicc
+for mpi in %{?mpi_list}; do
+  mkdir $mpi
+  pushd $mpi
+  %module_load $mpi
+  %configure --disable-static --includedir=%{mpi_includedir}/$mpi%{mpi_include_ext} --libdir=%{mpi_libdir}/$mpi/%{mpi_lib_ext}
+  %make_build
+  module purge
+  popd
+done
+
+%install
+for mpi in %{?mpi_list}; do
+  %module_load $mpi
+  %make_install -C $mpi
+  rm %{buildroot}/%{mpi_libdir}/$mpi/%{mpi_lib_ext}/*.la
+  rm -r %{buildroot}%{_datadir}/lwgrp
+  module purge
+done
+
+%if %{with_openmpi3}
+%files openmpi3
+%license LICENSE.TXT
+%doc README
+%if (0%{?suse_version} >= 1500)
+%files -n liblwgrp0-openmpi3
+%endif
+%{mpi_libdir}/openmpi3/%{mpi_lib_ext}/liblwgrp.so.*
+
+%files openmpi3-devel
+%{mpi_includedir}/openmpi3%{mpi_include_ext}/lwgrp.h
+%{mpi_libdir}/openmpi3/%{mpi_lib_ext}/liblwgrp.so
+%{mpi_libdir}/openmpi3/%{mpi_lib_ext}/pkgconfig/liblwgrp.pc
+%endif
+
+%if %{with_mpich}
+%files mpich
+%license LICENSE.TXT
+%doc README
+%if (0%{?suse_version} >= 1500)
+%files -n liblwgrp0-mpich
+%endif
+%{mpi_libdir}/mpich/%{mpi_lib_ext}/liblwgrp.so.*
+
+%files mpich-devel
+%{mpi_includedir}/mpich%{mpi_include_ext}/lwgrp.h
+%{mpi_libdir}/mpich/%{mpi_lib_ext}/liblwgrp.so
+%{mpi_libdir}/mpich/%{mpi_lib_ext}/pkgconfig/liblwgrp.pc
+%endif
+
+%changelog
+* Mon Sep 28 2020 Brian J. Murrell <brian.murrell@intel.com> - 1.0.2-3.2
+- Package for multiple MPI stacks and multiple distros
+
+* Tue Sep 22 2020 John E. Malmberg <john.e.malmberg@intel.com> - 1.0.2-3.1
+- Change to use openmpi3 for CentOS.
```

Comment:
As my stuff never got released, this can be combined back to 1.0.2-3.1

Reply:
The work still started with you, so I think it's fair to give credit even though it was never landed.